app/api/probe_dock/reports_api.rb

```diff
@@ -116,6 +122,31 @@ def serialization_options reports
             paginated_rel = paginated_rel.where('categories.name in (?)', params[:categoryNames].collect(&:to_s).to_a)
           end
 
+          if array_param?(:status) || array_param?(:kind)
+            where_clauses = []
+
+            if array_param?(:status)
+              where_clauses << 'test_payloads.passed_results_count > 0' if params[:status].include?('passed')
+              where_clauses << 'test_payloads.results_count - test_payloads.passed_results_count - test_payloads.inactive_results_count > 0' if params[:status].include?('failed')
+              where_clauses << 'test_payloads.inactive_results_count > 0' if params[:status].include?('inactive')
+            end
+
+            if array_param?(:kind)
+              where_clauses << 'test_payloads.tests_count - test_payloads.new_tests_count > 0' if params[:kind].include?('existing')
+              where_clauses << 'test_payloads.new_tests_count > 0' if params[:kind].include?('new')
+            end
+
+            complete_clause = ''
+            where_clauses.each_with_index do |clause, idx|
+              complete_clause += clause
+              if idx < where_clauses.size - 1
+                complete_clause += ' OR '
+              end
+            end
+
+            paginated_rel = paginated_rel.where(complete_clause)
```

Comment:
Using `where_clauses.join(' OR ')` here would allow you to remove the entire previous code block.